numpy/core/tests/test_arrayprint.py

```diff
@@ -237,12 +243,26 @@ def test_formatter_reset(self):
         assert_equal(repr(x), "array([ 0.,  1.,  2.])")
 
     def test_0d_arrays(self):
+        if sys.version_info[0] >= 3:
+            assert_equal(str(np.array('café', np.unicode_)), 'café')
+            assert_equal(repr(np.array('café', np.unicode_)),
+                         "array('café',\n      dtype='<U4')")
+            assert_equal(str(np.array('café', np.unicode_)), 'café')
+        else:
+            assert_equal(repr(np.array(u'café', np.unicode_)),
+                         "array(u'caf\\xe9',\n      dtype='<U4')")
+        assert_equal(str(np.array('test', np.str_)), 'test')
+
+        a = np.zeros(1, dtype=[('a', '<i4', (3,))])
+        assert_equal(str(a[0]), '([0, 0, 0],)')
+
         assert_equal(repr(np.datetime64('2005-02-25')[...]),
-                     "array('2005-02-25', dtype='datetime64[D]')")
+            "array(numpy.datetime64('2005-02-25'), dtype='datetime64[D]')")
```

Comment:
Although I think I understand why this happens, it does not seem very nice. And it is still a regression compared to 1.12:
```
np.datetime64('2005-02-25')
# numpy.datetime64('2005-02-25')
np.datetime64('2005-02-25')[...]
# array(datetime.date(2005, 2, 25), dtype='datetime64[D]')
```